frontend: Edit confirm_continue_registration.html to be clearer.

vaidap opened the pull request:
Fixes #5707.
The second commit is just for me for testing.

* Do we want to center the text?
* Made the second button redirect to `action="/"`, does this make sense?
![image](https://user-images.githubusercontent.com/13666710/28375300-067e6a26-6c9f-11e7-9b7c-14bb858b6a83.png)

rishig commented:
Thanks @vaidap!

How about: "No account found for {{ email }}. Would you like to register instead?"

And for styling, I think having both buttons on the same row would be nice. Maybe the "Go back" button should be the filled in CTA, and the the "Continue to registration" should use the outline styling of button?

vaidap commented:
It now looks like this: 
![image](https://user-images.githubusercontent.com/13666710/28411025-712c8c66-6d37-11e7-9d8b-676da92d7887.png)

Couldn't find a barebones just-the-outline button so styled it myself, let me know if there is an already written class i should be using?

timabbott commented:
Looks reasonable and certainly better. Do we want the action of "Go back" to be `/login`, not `/` (which will eventually redirect to `/login`)?  And maybe it should say "Go back to login"?

@rishig @brockwhittaker thoughts?

rishig commented:
cool, thanks @vaidap! 
* Let's remove the "?" from "Continue to Registration?"?
* I think the "No account found .." text should be bigger, but that's probably a more global change.
* Would adding some icon like http://fontawesome.io/icon/arrow-left/ to the back button make sense?

I guess the worry with "Go back" is that they just came from a google auth page, not the login page. Agree with "Go back to login" or "Back to login".

vaidap commented:
* Addressed the concerns-- this is what it looks like now: 
![image](https://user-images.githubusercontent.com/13666710/28490173-249a3fa2-6ecb-11e7-8988-30036beaf08e.png)
* Increasing the text font size makes the text wrap-- not sure if we want to decrease the container's size?
`600px` width:
![image](https://user-images.githubusercontent.com/13666710/28490174-324dc060-6ecb-11e7-9f0a-ae1ce64c51a0.png)
`580px` width:
![image](https://user-images.githubusercontent.com/13666710/28490199-bcf7ca94-6ecb-11e7-8bd8-3d264a223b0a.png)

lonerz commented:
Well, text is going to wrap depending on email length anyway, so we might want to think if we want everything centered or left-justified.

rishig commented:
cool, and thanks for all the screenshots! 
* I wonder if there a thinner font weight version of the back arrow?
* I would reduce the gap between the two buttons (closer to how you had it before)
* 580px width and wrapping sounds like a good option (though agree with @lonerz that it'll probably wrap differently depending on email length). The text should be left-justified, and we can make the decision of whether the title and buttons should be left justified later, I think (in the context of what we do with our other pages of this form)

rishig commented on templates/zerver/confirm_continue_registration.html:
```diff
@@ -23,28 +23,38 @@
 
                         {% endtrans %}
                     </p>
                     {# TODO: Ideally, this should use whatever auth #}
```
this comment should go after the "Go back to login" block (it refers to the "Continue to registration" block)

rishig commented:
@timabbott what do you think about changing "Continue to registration" to "Register instead"? If we can only get to this page via google or github auth, clicking that button should just dump them into the tutorial (i.e. there are no more portico pages needed).
(Currently it doesn't do that, but hopefully fixing this will be easy once Umair's work on not requiring a password is complete)

vaidap commented:
* put the two buttons closer:
![image](https://user-images.githubusercontent.com/13666710/28515972-7d4dbef2-7057-11e7-9251-636a899c98bb.png)

* regarding a thinner font version of the arrow, fontawesome doesn't provide that. there are a couple of workarounds:
```
    color: hsl(213, 23%, 25%);
    text-shadow:
    3px 3px 0 white,
   -1px -1px 0 white,  
    1px -1px 0 white,
   -1px  1px 0 white,
    1px  1px 0 white;
```
gives this effect:
![image](https://user-images.githubusercontent.com/13666710/28515663-5bd73fb0-7056-11e7-8c50-f57bad5b2a63.png)

`-webkit-text-stroke: 1px hsl(213, 23%, 25%);` gives the nicest effect, but only webkit supports it.
![image](https://user-images.githubusercontent.com/13666710/28515780-c4d787a4-7056-11e7-9238-f5a72378433c.png)

finally, making it smaller & moving it a bit more left looks like this: 
![image](https://user-images.githubusercontent.com/13666710/28515819-f1205886-7056-11e7-8f0d-849b338fed2f.png)

let me know which one makes most sense.

timabbott commented:
Some thoughts:
* The webkit one definitely looks nicest, sigh.  Let's just get rid of the arrow?  
* +1 for changing the right button text to "Register instead".  
* I think the two buttons should appear more centered in the page; it doesn't look quite right.

vaidap commented:
made the changes, i think the buttons seem not as centered because of the way i took the screenshot/the text in the paragraph above?
here's what it looks like now:
![image](https://user-images.githubusercontent.com/13666710/28768859-2e9cf500-75d1-11e7-9ba3-b5bdb92d3f1f.png)

with centered text:
![image](https://user-images.githubusercontent.com/13666710/28768876-3ef2de1a-75d1-11e7-90ff-05f3e9e3ca76.png)

setting the container back to `600px` so the text is in one line:
![image](https://user-images.githubusercontent.com/13666710/28768895-4b69325c-75d1-11e7-9324-26d29adafeb4.png)

rishig commented:
I think left justifying everything (header, text, buttons) could solve the problem.
Also, possibly the "No account found .." text could be a point or two larger?

vaidap commented:
made changes: here's what it looks like right now. had to make the paragraph font a bit lighter because it became more strongly bolded when increased in size.

![image](https://user-images.githubusercontent.com/13666710/28819697-f63de012-76a6-11e7-9c83-730e1c0b3fff.png)

brockwhittaker commented:
I would up the font weight on the paragraph, change the width to the same as the header, make sure the buttons are vertically centered compared to the \<h1> tag, and maybe change the text to like #aaa or whatever is okay for contrast.

vaidap commented:
made the changes @brockwhittaker !
![image](https://user-images.githubusercontent.com/13666710/28866704-a79b3804-776b-11e7-865a-c12b0a84cab3.png)

timabbott commented:
Looks great, merged, thanks @vaidap!

timabbott commented:
(I squashed the commits, because stylistically, updates to tests belong in the same commit as the change that required those test updates; otherwise tools like `git bisect` will fail on the intermediate commits)

vaidap commented:
Ah that makes sense, I'll make sure to work like that from now on!
